### PR TITLE
Remove support for PHP 5.6, allow > PHP 7.0, Add type hints and tests.

### PR DESCRIPTION
**Pull request by samgreenwood**

### CODE REVIEW

**Comment by chnorton** [comment]
@samgreenwood I couldn't see any tests in the changes - did these get left out of the commit?

**Comment by samgreenwood** [comment]
@chnorton whoops, forgot to add, should be there now.

File: phpunit.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- https://phpunit.de/manual/current/en/appendixes.configuration.html -->
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.8/phpunit.xsd"
```
Comment (mwgamble) [inline phpunit.xml]: Given we're using at least PHPUnit 6.4, this URL should probably refer to `6.4`, not `4.8`.

File: src/Generator.php [hunk 2]
```diff
@@ -1,4 +1,6 @@
 <?php
+declare(strict_types=1);
```
Comment (mwgamble) [inline src/Generator.php]: This should go below the license header. The file should look like this:

```
<?php
/**
 * <license header goes here>
 */

declare(strict_types=1);

namespace MyNamespace;

// Any use statements would go here

class MyClass
{
}

File: src/Generator.php [hunk 3]
```diff
@@ -27,12 +29,9 @@ class Generator
      * @param int $minimumLength
      * @return string
      */
-    public function calcMod10V1($number, $separator = "", $minimumLength = 6)
+    public function calcMod10V1(string $number, string $separator = "", int $minimumLength = 6) : string
```
Comment (mwgamble) [inline src/Generator.php]: There shouldn't be a space between the closing parenthesis and the colon.

File: src/Generator.php [hunk 4]
```diff
@@ -64,11 +63,8 @@ public function calcMod10V1($number, $separator = "", $minimumLength = 6)
      * @param string $number
      * @return string
      */
-    public function calcMod10V5($number)
+    public function calcMod10V5(string $number) : string
```
Comment (mwgamble) [inline src/Generator.php]: There shouldn't be a space between the closing parenthesis and the colon.

File: tests/GeneratorTest.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (mwgamble) [inline tests/GeneratorTest.php:1]: We should have strict typing turned on in this file.

File: tests/GeneratorTest.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+        $number = $generator->calcMod10V1('1234567890');
+
+        $this->assertSame('12345678903', $number);
+
+    }
+
+    public function testItGeneratesValidMod10V5()
+    {
+        $generator = new Generator();
```
Comment (mwgamble) [inline tests/GeneratorTest.php]: Given every test is going to want an instance of the `Generator` class, its instantiation should probably be moved to a `setUp` method.

**Review by mwgamble — CHANGES_REQUESTED** [review]
Thanks, your changes look good. There's just a few small code style issues that need fixing up and then it should be good to go.

File: phpunit.xml (hunk 1 above)
Comment (samgreenwood) [reply phpunit.xml]: Resolved.

File: src/Generator.php (hunk 2 above)
Comment (samgreenwood) [reply src/Generator.php]: Resolved.

File: src/Generator.php (hunk 3 above)
Comment (samgreenwood) [reply src/Generator.php]: Resolved.

File: src/Generator.php (hunk 4 above)
Comment (samgreenwood) [reply src/Generator.php]: Resolved.

File: tests/GeneratorTest.php (hunk 5 above)
Comment (samgreenwood) [reply tests/GeneratorTest.php:1]: Resolved.

File: src/Generator.php [hunk 7]
```diff
@@ -1,5 +1,4 @@
 <?php
-declare(strict_types=1);
```
Comment (mwgamble) [inline src/Generator.php]: There shouldn't be a blank line between the opening PHP tag and the license header, as per my previous comment.

File: tests/GeneratorTest.php [hunk 8]
```diff
@@ -1,25 +1,29 @@
...
+declare(strict_types=1);
+
 use PHPUnit\Framework\TestCase;
 use Fontis\BpayRefGenerator\Generator;
 
 class GeneratorTest extends TestCase
 {
+    private $generator;
+
+    public function setUp() {
```
Comment (mwgamble) [inline tests/GeneratorTest.php]: The opening brace should be on the next line.

File: tests/GeneratorTest.php [hunk 9]
```diff
@@ -1,25 +1,29 @@
 <?php
 
+declare(strict_types=1);
+
 use PHPUnit\Framework\TestCase;
 use Fontis\BpayRefGenerator\Generator;
```
Comment (mwgamble) [inline tests/GeneratorTest.php]: The list of `use` statements should be sorted alphabetically. Apologies for not brining this up in my previous set of comments.

File: tests/GeneratorTest.php [hunk 10]
```diff
@@ -1,25 +1,29 @@
...
 
 class GeneratorTest extends TestCase
 {
+    private $generator;
```
Comment (mwgamble) [inline tests/GeneratorTest.php]: This should have a docblock:

```
/**
 * @var MyClassName
 */
private $property;
```

**Review by mwgamble — CHANGES_REQUESTED** [review]
Thanks, there's just a few more changes required.

**Comment by samgreenwood** [comment]
Third times a charm ;)

**Review by mwgamble — APPROVED** [review]

**Comment by chnorton** [comment]
Thanks for making those further updates @samgreenwood - we'll merge in your changes shortly!

**Comment by mwgamble** [comment]
Thanks @samgreenwood!

**Comment by chnorton** [comment]
These changes have been merged in now as part of commit a9eaa536165d49a086445bf8a7effb7e1f978165 and tagged as 2.0. Thanks again @samgreenwood!

**Pull request closed without merging**